roles/openshift_common/tasks/atomic_docker_check.yml

```diff
@@ -0,0 +1,16 @@
+---
+# Fail early if Atomic and old version of Docker
+- block:
+
+  - name: Determine available Docker
+    shell: "rpm -q --queryformat '---\ncurr_version: %{VERSION}\n' docker"
+    register: l_atomic_docker_version_result
+
+  - set_fact:
+      l_atomic_docker_version: "{{ l_atomic_docker_version_result.stdout | from_yaml }}"
+
+  - fail:
+      msg: Installation on Atomic Host requires Docker 1.12 or later. Please upgrade and restart the Atomic Host.
+    when: l_atomic_docker_version.curr_version | version_compare('1.12','<')
```

Comment:
Instead of using `set_fact` to provide an intermediate step, I would suggest adding moving the definition of  `l_atomic_docker_version` to the task_vars. I also think that the `assert` module might be more appropriate here, especially since we have versions of OpenShift that would not be able to work with newer versions of docker as well.

```
- assert:
    msg: blah, blah, something incompatible verisons found
    that:
    - l_current_docker_version | version_compare('1.12', '>=')
    - l_current_docker_version | version_compare('1.13', '<') # hypothetically
  vars:
    l_current_docker_version: "{{ (l_atomic_docker_version_result.stdout | from_yaml).curr_version }}"
```

That would also make it a bit easier to add assertions for non-atomic hosts where we would need to compare against a combination of installed and available packages.

Downside to this is that it makes it harder to give a pretty error message, but I think it would be better to handle those types of issues in a callback plugin rather than having a ton of `fail` statements.